llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -4652,14 +4652,16 @@ bool CombinerHelper::matchReassocFoldConstantsInSubTree(GPtrAdd &MI,
   if (!C2)
     return false;
 
-  // If constant widths differ pick larger one.
-  unsigned BitWidth = std::max(C1->getBitWidth(), C2->getBitWidth());
-  APInt NewConst = C1->sext(BitWidth) + C2->sext(BitWidth);
-  LLT Type =
-      MRI.getType(C1->getBitWidth() > C2->getBitWidth() ? LHSSrc2 : Src2Reg);
+  // Pick correct size for the constant based on the address space
```

Comment:
If #84352 is accepted then this won't be necessary - you can rely on C1 and C2 both having the correct (same) width already.

Reply:
Ok then